.github/scripts/attack.py

```
@@ -336,8 +336,72 @@ def attack_test():
+def wait_for_broker_quiesce(timeout_sec: float = 60.0) -> bool:
+    """
+    Block until a fresh client completes a QoS-1 subscribe/publish/receive
+    round trip promptly, proving the broker has drained the stress backlog.
+    """
+    log = mk_logger("SETTLE")
+    deadline = time.time() + timeout_sec
+    while time.time() < deadline:
+        connected = threading.Event()
+        got = threading.Event()
+        c = new_client(f"SETTLE_{random.randint(1000, 9999)}")
+        c.on_connect = lambda cl, u, f, rc, p: connected.set()
```

Ruff (0.15.21)

[warning] Function definition does not bind loop variable `connected`

(B023)